Updates for Terraform policy docs (#2864)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/reference/policy/module-policy.mdx

The `module_policy` block defines policies for your workspaces' Terraform modules. Module policies allow you to set guardrails at the abstraction layer where infrastructure is often packaged and shared. The first label of the block specifies the module source. The second label is a unique name for the policy itself. You can use a wildcard (*) for the module source to apply the policy to every module in the workspace.

Terraform modules define configuration for one or more resources. Use [resource policies](/terraform/policy/reference/policy/resource-policy) to target resources within modules.

@include "beta.mdx"

## Configuration model

The `module_policy` block supports the following configuration:

- `module_policy` &nbsp "[`<module_source>`](#module-source)" &nbsp "[`<policy_label>`](#policy-label)" &nbsp block | repeatable
  - [`enforce`](#enforce) &nbsp block | repeatable
    - [`condition`](#condition) &nbsp boolean expression | required
    - [`error_message`](#error-message) &nbsp string | optional
    - [`info_message`](#info-message) &nbsp string | optional

>   - [`enforcement_level`](#enforcement-level) &nbsp string | optional

Reviewer: ```suggestion
    - [`error_message`](#error_message) &nbsp string | optional
    - [`info_message`](#info_message) &nbsp string | optional
  - [`enforcement_level`](#enforcement_level) &nbsp string | optional
```

  - [`filter`](#filter) &nbsp boolean expression | optional
  - [`locals`](#locals) &nbsp block | optional

>     - `<name>` &nbsp any | repeatable

Reviewer: ```suggestion
    - [`<name>`](#locals) &nbsp any | repeatable
```

## Complete configuration example

> The following example demonstrates a module policy block that matches all modules with all configuration options specified, and a module policy block that matches a specified module:

Reviewer: ```suggestion
The following example demonstrates a `module_policy` block that matches all modules with all configuration options specified and a `module_policy` block that matches a specified module:
```

<CodeBlockConfig filename="policies/example.policy.hcl">

```hcl
module_policy "*" "version_and_source_check" {
  enforcement_level = "mandatory"
  filter            = meta.version != null

  locals {
    min_version = "1.0.0"
  }

  enforce {

>     condition     = core::contains(local.approved_sources, meta.source)

Reviewer 2: This enforce block references `local.approved_sources`, but the `locals` block above only defines `min_version` (the top-level `locals` with `approved_sources` was removed in this restructure), so the example would fail evaluation. Either add `approved_sources` to the policy-scoped `locals` or change this condition to use another check.

}
```

</CodeBlockConfig>

## Specification

A `module_policy` block supports the following configuration.

### Module source

Specifies where Terraform policy retrieves the module source code. Refer to the [Terraform module block reference](/terraform/language/block/module#source) for valid module sources.

Use `*` to match all modules.

### Policy label

A unique local name for this policy within the policy file.

### `enforce`

> The `enforce` block defines a condition that must evaluate to `true` for the policy to pass. You can include one or more `enforce` blocks in each `module_policy` block. If any of the enforce blocks' conditions fail during evaluation, then Terraform policy will fail the policy and return the corresponding error messages.

Reviewer: ```suggestion
The `enforce` block defines a condition that must evaluate to `true` for the policy to pass. You can include one or more `enforce` blocks in each `module_policy` block. If any of the enforce blocks' conditions fail during evaluation, then Terraform policy fails the policy and returns the corresponding error messages.
```

- Data type: Block
- Repeatable

An `enforce` block supports the following configuration.

#### `condition`

A boolean expression that must evaluate to `true` for the policy to pass.

- Data type: Boolean expression
- Required

#### `error_message`

The error message to display when the condition evaluates to `false`.

- Data type: String
- Default: None

#### `info_message`

An informational message to display regardless of whether the policy passes or fails.

- Data type: String
- Default: None

### `enforcement_level`

Specifies the enforcement level for the policy. You can specify one of the following values:

> - `"advisory"`: Allows the operation to continue but prints a warning when if the conditions of the policy aren't met.

Reviewer 2: Typo: "prints a warning when if the conditions" -- drop "if" (same wording also appears in `provider-policy.mdx` and `resource-policy.mdx`).
